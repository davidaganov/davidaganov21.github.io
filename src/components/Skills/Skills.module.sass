.skills
  @media (min-width: 1000px)
    margin-top: 8rem
    padding-top: 3rem
  @media (max-width: 999px)
    margin-top: 5rem
    padding-top: 1.5rem

.accordion
  margin-top: 5rem

.item
  $parent: &
  position: relative
  padding-bottom: 1rem
  color: var(--color-white)
  &::before
    position: absolute
    content: ""
    right: 0
    bottom: 0
    width: 100%
    height: .1rem
    background-color: var(--color-accent-300)
  &:not(:last-of-type)
    margin-bottom: 2rem
  span
    display: flex
    align-items: center
    justify-content: right
    color: var(--color-white)
    cursor: pointer
    &::before
      display: inline-block
      content: ""
      width: 2rem
      height: 2rem
      color: tomato
      background-repeat: no-repeat
      background-position: center
      background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24'%3E%3Cpath fill='%23B87EEF' d='M9.75 3v6.75H3v4.5h6.75V21h4.5v-6.75H21v-4.5h-6.75V3h-4.5Z'/%3E%3C/svg%3E")
      transition: transform .2s
    &:focus-visible
      color: var(--color-accent)
      text-decoration: underline
    h3
      display: flex
      align-items: center
      margin-left: auto
      @media (min-width: 576px)
        font: bold 2.4rem var(--main-font)
      @media (max-width: 575px)
        font: bold 2rem var(--main-font)
    svg
      margin-left: 2.5rem

.list
  position: relative
  display: flex
  flex-direction: column
  margin: 3rem 2rem 2rem 0
  padding-right: 2.5rem
  text-align: right
  &::after
    position: absolute
    content: ""
    right: 0
    top: 0
    height: 100%
    width: .1rem
    background-color: var(--color-accent)

.content
  &:not(:last-of-type)
    margin-bottom: 2.2rem

.title
  font: 600 2rem var(--main-font)

.listNested
  display: flex
  justify-content: flex-end

.listNestedItem
  margin-top: .55rem
  font: normal 1.8rem var(--main-font)
  color: var(--color-white-700)
  &:not(:last-of-type)
    margin-right: .7rem
    &::after
      content: ", "

.empty
  margin-top: 5rem
  text-align: right
  font: normal 2rem var(--second-font)
  color: var(--color-white-700)
