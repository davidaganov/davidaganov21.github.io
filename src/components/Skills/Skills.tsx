import styles from "./Skills.module.sass"
import cn from "classnames"

import Collapsible from "react-collapsible"
import { Title } from "../"
import data from "../../data"

export const Skills = (): JSX.Element => {
  const { skills } = data

  const buildSkills = () => {
    const categories = skills.map(({ id, title, list }) => {
      return (
        <Collapsible
          trigger={title}
          classParentString={styles.item}
          triggerElementProps={{ id: `skill-${id}` }}
          contentElementId={`skill-${id}`}
          transitionTime={350}
          easing={"ease-in-out"}
          key={id}
          tabIndex={0}
        >
          <ul className={styles.list}>{buildItems(list)}</ul>
        </Collapsible>
      )
    })

    return <div className={styles.accordion}>{categories}</div>
  }

  const buildItems = (list: { id: string; title: string; tag: string[] }[]) => {
    return list.map(({ id, title, tag }) => {
      return (
        <li
          className={styles.content}
          key={id}
        >
          <h3 className={styles.title}>{title}</h3>
          <ul className={styles.listNested}>{buildTags(tag)}</ul>
        </li>
      )
    })
  }

  const buildTags = (tag: string[]) => {
    return tag.map((skill) => {
      return (
        <li
          className={styles.listNestedItem}
          key={skill}
        >
          {skill}
        </li>
      )
    })
  }

  return (
    <section
      className={cn(styles.skills, "section-rtl")}
      id="skills"
    >
      <div className="inner">
        <Title
          number={2}
          link="#skills"
          title="My Skills"
          direction="rtl"
        />

        {buildSkills()}
      </div>
    </section>
  )
}
