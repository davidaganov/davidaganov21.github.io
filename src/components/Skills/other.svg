<svg xmlns="http://www.w3.org/2000/svg" fill="none" width="40" height="40" viewBox="0 0 24 24">
  <path fill="#B87EEF" d="M12 3a9.01 9.01 0 0 0 0 18 9.01 9.01 0 0 0 0-18ZM7.68 13.44a1.44 1.44 0 1 1-.06-2.88 1.44 1.44 0 0 1 .06 2.88Zm4.32 0a1.44 1.44 0 1 1-.06-2.88 1.44 1.44 0 0 1 .06 2.88Zm4.32 0a1.44 1.44 0 1 1-.06-2.88 1.44 1.44 0 0 1 .06 2.88Z"/>
</svg>
