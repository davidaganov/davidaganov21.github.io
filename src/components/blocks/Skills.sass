.skills
  @media (min-width: $middle-min)
    margin-top: 8rem
    padding-top: 3rem
  @media (max-width: $middle-max)
    margin-top: 4rem
    padding-top: 1.5rem

  &__accordion
    margin-top: 6rem

  &-item
    $parent: &
    position: relative
    padding-bottom: 1rem
    color: var(--color-white)
    &::before
      position: absolute
      content: ""
      right: 0
      bottom: 0
      width: 100%
      height: .1rem
      background-color: var(--color-accent-300)
    &:not(:last-of-type)
      margin-bottom: 2rem
    &:nth-of-type(1) #{$parent}__trigger::after
      background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 20 24'%3E%3Cpath fill='%23B87EEF' d='M18.18 21.2 20 0H0l1.82 21.2L9.97 24l8.21-2.8ZM4.83 17.45l-.31-4.06H7l.19 2.04 2.78.77 2.8-.77.31-3.33h-8.7l-.66-7.8h12.56l-.23 2.55H6.48l.21 2.65h9.15l-.71 7.95-5.1 1.44v.02h-.06l-5.14-1.46Z'/%3E%3C/svg%3E")
    &:nth-of-type(2) #{$parent}__trigger::after
      background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 20 24'%3E%3Cpath fill='%23B87EEF' d='m0 0 1.82 21.6L10 24l8.18-2.4L20 0H0Zm16.1 7.06-.22 2.72-.7 8.13-5.18 1.5-5.18-1.5-.36-4.25h2.53l.2 2.2 2.8.8H10l2.82-.8.38-3.67-8.87.02-.25-2.51 9.34-.12.18-2.69-9.77.03-.16-2.5h12.66l-.22 2.64Z'/%3E%3C/svg%3E")
    &:nth-of-type(3) #{$parent}__trigger::after
      background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 20 24'%3E%3Cpath fill='%23B87EEF' d='M1.82 21.6 0 0h20l-1.82 21.6L10 24l-8.18-2.4Zm2.67-7.89.35 4.17L10 19.4l5.13-1.5.69-8.14h-8.6l.35-4.14 2.43-.1h.45l2.56.1.24 2.8h2.51l-.48-5.66H5.33l-.8 9.64h8.53l-.3 3.43-2.76.8-2.79-.79-.2-2.13H4.48Z'/%3E%3C/svg%3E")
    &:nth-of-type(4) #{$parent}__trigger::after
      background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24'%3E%3Cpath fill='%23B87EEF' d='M12 3a9.01 9.01 0 0 0 0 18 9.01 9.01 0 0 0 0-18ZM7.68 13.44a1.44 1.44 0 1 1-.06-2.88 1.44 1.44 0 0 1 .06 2.88Zm4.32 0a1.44 1.44 0 1 1-.06-2.88 1.44 1.44 0 0 1 .06 2.88Zm4.32 0a1.44 1.44 0 1 1-.06-2.88 1.44 1.44 0 0 1 .06 2.88Z'/%3E%3C/svg%3E")
    &__trigger
      display: flex
      align-items: center
      justify-content: right
      font: bold 2.4rem var(--main-font)
      cursor: pointer
      &::before, &::after
        display: inline-block
        content: ""
        background-repeat: no-repeat
        background-position: center
      &::before
        width: 2rem
        height: 2rem
        margin-right: auto
        background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24'%3E%3Cpath fill='%23B87EEF' d='M9.75 3v6.75H3v4.5h6.75V21h4.5v-6.75H21v-4.5h-6.75V3h-4.5Z'/%3E%3C/svg%3E")
        transition: transform .2s
      &::after
        width: 4rem
        height: 4rem
        margin-left: 2.5rem
      &:focus-visible
        color: var(--color-accent)
        text-decoration: underline
      &.is-open::before
        transform: rotate(45deg)

  &__list
    position: relative
    display: flex
    flex-direction: column
    margin: 3rem 2rem 2rem 0
    padding-right: 2.5rem
    text-align: right
    &::after
      position: absolute
      content: ""
      right: 0
      top: 0
      height: 100%
      width: .1rem
      background-color: var(--color-accent)
  &__content
    &-title
      font: 600 2rem var(--main-font)
    &:not(:last-of-type)
      margin-bottom: 2.2rem

  &__list-nested
    display: flex
    justify-content: flex-end
  &__item
    margin-top: .55rem
    font: normal 1.8rem var(--main-font)
    color: var(--color-white-700)
    &:not(:last-of-type)
      margin-right: .7rem
      &::after
        content: ", "
