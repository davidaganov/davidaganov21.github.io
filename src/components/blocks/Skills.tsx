import Collapsible from "react-collapsible"
import { v4 as uuidv4 } from "uuid"

export default function Skills() {
  const data = [
    {
      id: uuidv4(),
      title: "HTML",
      list: [
        { id: uuidv4(), title: "Template Engine", tag: ["PUG"] },
        { id: uuidv4(), title: "Native", tag: ["HTML5"] }
      ]
    },
    {
      id: uuidv4(),
      title: "CSS",
      list: [
        { id: uuidv4(), title: "Preprocessors", tag: ["SASS/SCSS", "Less"] },
        { id: uuidv4(), title: "Frameworks", tag: ["Bootstrap", "Tailwind"] },
        { id: uuidv4(), title: "Native", tag: ["CSS3"] }
      ]
    },
    {
      id: uuidv4(),
      title: "JavaScript",
      list: [
        { id: uuidv4(), title: "Frameworks", tag: ["Nuxt.js", "Vue.js"] },
        { id: uuidv4(), title: "Library", tag: ["React", "React Native", "JQuery"] },
        { id: uuidv4(), title: "Native", tag: ["JavaScript / TypeScript"] }
      ]
    },
    {
      id: uuidv4(),
      title: "Other",
      list: [
        { id: uuidv4(), title: "CMS", tag: ["Wordpress"] },
        { id: uuidv4(), title: "Design", tag: ["Figma", "Adobe Photoshop"] },
        { id: uuidv4(), title: "3D", tag: ["Blender", "Maya"] },
        { id: uuidv4(), title: "OS", tag: ["Windows", "Mac OS"] }
      ]
    }
  ]

  const renderSkills = (
    arr: { id: string; title: string; list: { id: string; title: string; tag: string[] }[] }[]
  ) => {
    const categories = arr.map(({ id, title, list }) => {
      const items = list.map(({ id, title, tag }) => {
        const tags = tag.map((skill) => {
          return (
            <li
              className="skills__item"
              key={skill}
            >
              {skill}
            </li>
          )
        })
        return (
          <li
            className="skills__content"
            key={id}
          >
            <h3 className="skills__content-title">{title}</h3>
            <ul className="skills__list-nested">{tags}</ul>
          </li>
        )
      })
      return (
        <Collapsible
          trigger={title}
          classParentString="skills-item"
          transitionTime={200}
          key={id}
        >
          <ul className="skills__list">{items}</ul>
        </Collapsible>
      )
    })

    return <div className="skills__accordion">{categories}</div>
  }

  return (
    <section
      className="skills section-rtl"
      id="skills"
    >
      <div className="skills__inner inner">
        <h2 className="skills__title section-title">
          <a href="#skills">
            <span>02.</span> My Skills
          </a>
        </h2>

        {renderSkills(data)}
      </div>
    </section>
  )
}
