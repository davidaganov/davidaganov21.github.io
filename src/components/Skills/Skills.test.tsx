import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import { Skills } from "./Skills"

const randomUUID = () => {
  let value = 0
  return () => value++
}

jest.mock("uuid", () => {
  return { v4: randomUUID() }
})

describe("Skills component", () => {
  it("Skills renders", () => {
    render(<Skills />)
    const headers = screen.getAllByRole("button").length
    const regions = screen.getAllByRole("region").length

    expect(headers === regions).toBeTruthy()
    expect(screen.getByText("#").closest("a")).toHaveAttribute("href", "#skills")
  })

  it("Skills snapshot", () => {
    const skills = render(<Skills />)

    expect(skills).toMatchSnapshot()
  })
})
