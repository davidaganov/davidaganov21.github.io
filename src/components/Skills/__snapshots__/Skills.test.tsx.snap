// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills component Skills snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="skills"
        id="skills"
      >
        <div
          class="inner"
        >
          <div
            class="title rtl"
          >
            <a
              aria-label="My Skills"
              href="#skills"
            >
              #
            </a>
            <h2>
              My Skills
               
              <span
                aria-hidden="true"
              >
                0
                2
                .
              </span>
            </h2>
          </div>
          <div
            class="accordion"
          >
            <div
              class="item"
            >
              <span
                aria-controls="content-4"
                aria-disabled="false"
                aria-expanded="false"
                class="item__trigger is-closed"
                id="trigger-4"
                role="button"
                tabindex="0"
              >
                <h3>
                  HTML
                  <svg>
                    html.svg
                  </svg>
                </h3>
              </span>
              <div
                aria-labelledby="trigger-4"
                class="item__contentOuter"
                id="content-4"
                role="region"
                style="height: 0px; transition: height 350ms ease-in-out; overflow: hidden;"
              >
                <div
                  class="item__contentInner"
                >
                  <ul
                    class="list"
                  >
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        Template Engine
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          PUG
                        </li>
                      </ul>
                    </li>
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        Other
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          HTML5
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="item"
            >
              <span
                aria-controls="content-5"
                aria-disabled="false"
                aria-expanded="false"
                class="item__trigger is-closed"
                id="trigger-5"
                role="button"
                tabindex="0"
              >
                <h3>
                  CSS
                  <svg>
                    css.svg
                  </svg>
                </h3>
              </span>
              <div
                aria-labelledby="trigger-5"
                class="item__contentOuter"
                id="content-5"
                role="region"
                style="height: 0px; transition: height 350ms ease-in-out; overflow: hidden;"
              >
                <div
                  class="item__contentInner"
                >
                  <ul
                    class="list"
                  >
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        Preprocessors
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          SASS/SCSS
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          Less
                        </li>
                      </ul>
                    </li>
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        Frameworks
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          Bootstrap
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          Tailwind
                        </li>
                      </ul>
                    </li>
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        Other
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          CSS3
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="item"
            >
              <span
                aria-controls="content-6"
                aria-disabled="false"
                aria-expanded="false"
                class="item__trigger is-closed"
                id="trigger-6"
                role="button"
                tabindex="0"
              >
                <h3>
                  JavaScript
                  <svg>
                    js.svg
                  </svg>
                </h3>
              </span>
              <div
                aria-labelledby="trigger-6"
                class="item__contentOuter"
                id="content-6"
                role="region"
                style="height: 0px; transition: height 350ms ease-in-out; overflow: hidden;"
              >
                <div
                  class="item__contentInner"
                >
                  <ul
                    class="list"
                  >
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        Frameworks
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          Nuxt.js
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          Vue.js
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          Next.js
                        </li>
                      </ul>
                    </li>
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        Library
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          React
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          React Native
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          JQuery
                        </li>
                      </ul>
                    </li>
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        Other
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          JavaScript
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          TypeScript
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="item"
            >
              <span
                aria-controls="content-7"
                aria-disabled="false"
                aria-expanded="false"
                class="item__trigger is-closed"
                id="trigger-7"
                role="button"
                tabindex="0"
              >
                <h3>
                  Other
                  <svg>
                    other.svg
                  </svg>
                </h3>
              </span>
              <div
                aria-labelledby="trigger-7"
                class="item__contentOuter"
                id="content-7"
                role="region"
                style="height: 0px; transition: height 350ms ease-in-out; overflow: hidden;"
              >
                <div
                  class="item__contentInner"
                >
                  <ul
                    class="list"
                  >
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        CMS
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          Wordpress
                        </li>
                      </ul>
                    </li>
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        Design
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          Figma
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          Adobe Photoshop
                        </li>
                      </ul>
                    </li>
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        3D
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          Blender
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          Maya
                        </li>
                      </ul>
                    </li>
                    <li
                      class="content"
                    >
                      <p
                        class="title"
                      >
                        OS
                      </p>
                      <ul
                        class="listNested"
                      >
                        <li
                          class="listNestedItem"
                        >
                          Windows
                        </li>
                        <li
                          class="listNestedItem"
                        >
                          Mac OS
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="skills"
      id="skills"
    >
      <div
        class="inner"
      >
        <div
          class="title rtl"
        >
          <a
            aria-label="My Skills"
            href="#skills"
          >
            #
          </a>
          <h2>
            My Skills
             
            <span
              aria-hidden="true"
            >
              0
              2
              .
            </span>
          </h2>
        </div>
        <div
          class="accordion"
        >
          <div
            class="item"
          >
            <span
              aria-controls="content-4"
              aria-disabled="false"
              aria-expanded="false"
              class="item__trigger is-closed"
              id="trigger-4"
              role="button"
              tabindex="0"
            >
              <h3>
                HTML
                <svg>
                  html.svg
                </svg>
              </h3>
            </span>
            <div
              aria-labelledby="trigger-4"
              class="item__contentOuter"
              id="content-4"
              role="region"
              style="height: 0px; transition: height 350ms ease-in-out; overflow: hidden;"
            >
              <div
                class="item__contentInner"
              >
                <ul
                  class="list"
                >
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      Template Engine
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        PUG
                      </li>
                    </ul>
                  </li>
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      Other
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        HTML5
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="item"
          >
            <span
              aria-controls="content-5"
              aria-disabled="false"
              aria-expanded="false"
              class="item__trigger is-closed"
              id="trigger-5"
              role="button"
              tabindex="0"
            >
              <h3>
                CSS
                <svg>
                  css.svg
                </svg>
              </h3>
            </span>
            <div
              aria-labelledby="trigger-5"
              class="item__contentOuter"
              id="content-5"
              role="region"
              style="height: 0px; transition: height 350ms ease-in-out; overflow: hidden;"
            >
              <div
                class="item__contentInner"
              >
                <ul
                  class="list"
                >
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      Preprocessors
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        SASS/SCSS
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        Less
                      </li>
                    </ul>
                  </li>
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      Frameworks
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        Bootstrap
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        Tailwind
                      </li>
                    </ul>
                  </li>
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      Other
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        CSS3
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="item"
          >
            <span
              aria-controls="content-6"
              aria-disabled="false"
              aria-expanded="false"
              class="item__trigger is-closed"
              id="trigger-6"
              role="button"
              tabindex="0"
            >
              <h3>
                JavaScript
                <svg>
                  js.svg
                </svg>
              </h3>
            </span>
            <div
              aria-labelledby="trigger-6"
              class="item__contentOuter"
              id="content-6"
              role="region"
              style="height: 0px; transition: height 350ms ease-in-out; overflow: hidden;"
            >
              <div
                class="item__contentInner"
              >
                <ul
                  class="list"
                >
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      Frameworks
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        Nuxt.js
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        Vue.js
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        Next.js
                      </li>
                    </ul>
                  </li>
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      Library
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        React
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        React Native
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        JQuery
                      </li>
                    </ul>
                  </li>
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      Other
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        JavaScript
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        TypeScript
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="item"
          >
            <span
              aria-controls="content-7"
              aria-disabled="false"
              aria-expanded="false"
              class="item__trigger is-closed"
              id="trigger-7"
              role="button"
              tabindex="0"
            >
              <h3>
                Other
                <svg>
                  other.svg
                </svg>
              </h3>
            </span>
            <div
              aria-labelledby="trigger-7"
              class="item__contentOuter"
              id="content-7"
              role="region"
              style="height: 0px; transition: height 350ms ease-in-out; overflow: hidden;"
            >
              <div
                class="item__contentInner"
              >
                <ul
                  class="list"
                >
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      CMS
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        Wordpress
                      </li>
                    </ul>
                  </li>
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      Design
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        Figma
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        Adobe Photoshop
                      </li>
                    </ul>
                  </li>
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      3D
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        Blender
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        Maya
                      </li>
                    </ul>
                  </li>
                  <li
                    class="content"
                  >
                    <p
                      class="title"
                    >
                      OS
                    </p>
                    <ul
                      class="listNested"
                    >
                      <li
                        class="listNestedItem"
                      >
                        Windows
                      </li>
                      <li
                        class="listNestedItem"
                      >
                        Mac OS
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
